runtime/model-protobuf/src/test/java/io/aklivity/zilla/runtime/model/protobuf/internal/ProtobufParserTest.java

```diff
@@ -0,0 +1,267 @@
+        // Given: Null input
+        ProtobufParser parser = new ProtobufParser(new FileDescriptor[0]);
+
+        // When: Parsing null string
+        parser.parse((String) null);
+
+        // Then: Should throw exception
+    }
+
+    @Test(expected = ProtobufParser.ProtobufParseException.class)
+    public void shouldThrowExceptionForEmptySchema()
+    {
+        // Given: Empty schema
+        String emptySchema = "";
+
+        // When: Parsing empty schema
+        ProtobufParser parser = new ProtobufParser(new FileDescriptor[0]);
+        parser.parse(emptySchema);
+
+        // Then: Should throw exception
+    }
+
+    @Test
+    public void shouldParseComplexProto2Schema()
+    {
+        // Given: A complex proto2 schema with enums, nested messages, and oneofs
+        String complexSchema =
+                "syntax = \"proto2\";\n" +
+                        "package complex;\n" +
+                        "\n" +
+                        "enum Status {\n" +
+                        "  UNKNOWN = 0;\n" +
+                        "  ACTIVE = 1;\n" +
+                        "  INACTIVE = 2;\n" +
+                        "}\n" +
+                        "\n" +
+                        "message Address {\n" +
+                        "  optional string street = 1;\n" +
+                        "  optional string city = 2;\n" +
+                        "}\n" +
+                        "\n" +
+                        "message Person {\n" +
+                        "  required string name = 1;\n" +
+                        "  optional int32 age = 2;\n" +
+                        "  optional Status status = 3;\n" +
+                        "  repeated Address addresses = 4;\n" +
+                        "  \n" +
+                        "  oneof contact {\n" +
+                        "    string email = 5;\n" +
+                        "    string phone = 6;\n" +
+                        "  }\n" +
+                        "}\n";
+
+        // When: Parsing the complex schema
+        ProtobufParser parser = new ProtobufParser(new FileDescriptor[0]);
+        FileDescriptor descriptor = parser.parse(complexSchema);
+
+        // Then: All elements should be correctly parsed
+        assertNotNull(descriptor);
+        assertEquals("proto2", descriptor.getSyntax());
+        assertEquals("complex", descriptor.getPackage());
+        assertEquals(2, descriptor.getMessageTypes().size());
+        assertEquals(1, descriptor.getEnumTypes().size());
+
+        // Verify enum
+        assertEquals("Status", descriptor.getEnumTypes().get(0).getName());
+        assertEquals(3, descriptor.getEnumTypes().get(0).getValues().size());
+
+        // Verify Person message with oneof
+        Descriptor personDescriptor = descriptor.findMessageTypeByName("Person");
+        assertNotNull(personDescriptor);
+        assertEquals(1, personDescriptor.getOneofs().size());
+        assertEquals("contact", personDescriptor.getOneofs().get(0).getName());
+    }
```

Comment:
Perhaps easier to just have the complex case test, but both v2 and v3, not just v2.